drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -188,10 +188,12 @@ int nxp_wifi_wlan_event_callback(enum wlan_event_reason reason, void *data)
 	case WLAN_REASON_CONNECT_FAILED:
 		net_eth_carrier_off(g_mlan.netif);
 		LOG_WRN("WLAN: connect failed");
+		wifi_mgmt_raise_connect_result_event(g_mlan.netif, -1);
 		break;
 	case WLAN_REASON_NETWORK_NOT_FOUND:
 		net_eth_carrier_off(g_mlan.netif);
 		LOG_WRN("WLAN: nxp_wlan_network not found");
+		wifi_mgmt_raise_connect_result_event(g_mlan.netif, -1);
```

Comment:
you can use status codes https://github.com/zephyrproject-rtos/zephyr/blob/f51729aadae9a7413290ee6417fb13b18d417147/include/zephyr/net/wifi_mgmt.h#L593